File: httpserver/Method.hs
module Method (Method(..), fromString) where

import Data.Char

data Method = Get 
			| Post
			| Put
			| Delete	
			deriving (Eq, Show)

fromString :: String -> Method
fromString s = 
	case map toUpper s of 
		"GET"    -> Get
		"POST"   -> Post
		"PUT"    -> Put
		"DELETE" -> Delete
	

File: rust_http_server/http/method.rs
#[derive(Show, PartialEq)]
pub enum Method {
	GET, POST, PUT, DELETE
}

impl Method {
	pub fn from_utf8(v: Vec<u8>) -> Option<Method> {
		match String::from_utf8(v) {
			Ok(s) => match s.as_slice() {
				"GET" => Some(Method::GET),
				"POST" => Some(Method::POST),
				"PUT" => Some(Method::PUT),
				_ => None
			},
			Err(s) => None
		}
	}
}